api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/AvailabilityHandler.java

```diff
@@ -0,0 +1,335 @@
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.wordnik.swagger.annotations.Api;
+import com.wordnik.swagger.annotations.ApiOperation;
+import com.wordnik.swagger.annotations.ApiParam;
+import com.wordnik.swagger.annotations.ApiResponse;
+import com.wordnik.swagger.annotations.ApiResponses;
+
+/**
+ * @author Stefan Negrea
+ *
+ */
+@Path("/")
+@Consumes(APPLICATION_JSON)
+@Produces(APPLICATION_JSON)
+@Api(value = "", description = "Availability metrics interface")
+public class AvailabilityHandler {
+    private static final long EIGHT_HOURS = MILLISECONDS.convert(8, HOURS);
+
+    @Inject
+    private MetricsService metricsService;
+
+    @POST
+    @Path("/{tenantId}/availability")
+    @ApiOperation(value = "Create availability metric definition. Same notes as creating numeric metric apply.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Metric definition created successfully"),
+            @ApiResponse(code = 400, message = "Missing or invalid payload", response = ApiError.class),
+            @ApiResponse(code = 409, message = "Availability metric with given id already exists",
+                response = ApiError.class),
+            @ApiResponse(code = 500, message = "Metric definition creation failed due to an unexpected error",
+                response = ApiError.class) })
+    public void createAvailabilityMetric(@Suspended final AsyncResponse asyncResponse,
+            @PathParam("tenantId") String tenantId, @ApiParam(required = true) AvailabilityMetric metric,
+            @Context UriInfo uriInfo) {
+        if (metric == null) {
+            Response response = Response.status(Status.BAD_REQUEST).entity(new ApiError("Payload is empty")).build();
+            asyncResponse.resume(response);
+            return;
+        }
+        metric.setTenantId(tenantId);
+        ListenableFuture<Void> future = metricsService.createMetric(metric);
+        URI created = uriInfo.getBaseUriBuilder().path("/{tenantId}/availability/{id}")
+                .build(tenantId, metric.getId().getName());
+        MetricCreatedCallback metricCreatedCallback = new MetricCreatedCallback(asyncResponse, created);
+        Futures.addCallback(future, metricCreatedCallback);
+    }
+
+    @GET
+    @Path("/{tenantId}/availability/{id}/tags")
+    @ApiOperation(value = "Retrieve tags associated with the metric definition.", response = Map.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Metric's tags were successfully retrieved."),
+            @ApiResponse(code = 204, message = "Query was successful, but no metrics were found."),
+            @ApiResponse(code = 500, message = "Unexpected error occurred while fetching metric's tags.",
+                response = ApiError.class) })
+    public void getAvailabilityMetricTags(@Suspended final AsyncResponse asyncResponse,
```

Comment:
This is fine solution also, as long as it is possible to request metadata somehow (which it currently isn't). The latter url sounds better to me (unless we want to provide datapoints from there).